script.rb

```diff
@@ -0,0 +1,271 @@
+
+    arrayed = []
+
+    if val.class == Regexp
+      my_each { |i| arrayed << val.match?(i) }
+    elsif block_given?
+      my_each { |i| arrayed << yield(i) }
+
+    elsif !val.nil? && !(val.instance_of? Integer) && !(val.instance_of? String)
+      my_each { |i| arrayed << (i.is_a? val ? true : false) }
+
+    elsif !val
+      my_each do |i|
+        arrayed << true if i.nil? || i == false
+      end
+
+    elsif val.instance_of? Integer
+      my_each { |i| arrayed << (i == val) }
+
+    elsif val.instance_of? String
+      my_each { |i| arrayed << (i == val) }
+
+    end
+
+    state = false
+
+    j = 0
+
+    while j < arrayed.size
+      if arrayed[j] == true
+        state = true
+        break
+      end
+      j += 1
+    end
+
+    state
+  end
+
+  def my_none?(val = nil)
+    return true if size.zero?
+
+    arrayed = []
+
+    if val.class == Regexp
+      my_each { |i| arrayed << val.match?(i) }
+    elsif block_given?
+      my_each { |i| arrayed << yield(i) }
+
+    elsif !val.nil? && !(val.instance_of? Integer) && !(val.instance_of? String)
+      my_each { |i| arrayed << (i.is_a? val ? true : false) }
+
+    elsif !val
+      my_each do |i|
+        if i.nil? || i == false
+          arrayed << false
+        elsif i == true
+          arrayed << true
+        end
+      end
+
+    elsif val.instance_of? Integer
+      my_each { |i| arrayed << (i == val) }
+
+    elsif val.instance_of? String
+      my_each { |i| arrayed << (i == val) }
+
+    end
+
+    state = true
+    j = 0
+
+    while j < arrayed.size
+      if arrayed[j] == true
+        state = false
+        break
+
+      end
+      j += 1
+
+    end
+
+    state
+  end
+
+  def my_count(val = 1)
+    num = 0
+
+    if val == 1 && block_given?
+
+      arrayed = []
+      my_each { |i| arrayed << i if yield(i) == true }
+
+      num = arrayed.size
+
+    elsif val == 1
+      track = 0
+
+      i = 0
+
+      while i < size
+        track += 1
+        i += 1
+      end
+
+      num = track
+      puts num
+
+    else
+
+      arrayed = my_select { |ent| ent == val }
+
+      num = arrayed.size
+    end
+
+    num
+  end
+
+  def my_map
+    return enum_for(:my_map) unless block_given?
+
+    arrayed = []
+
+    i = 0
+
+    while i < size
+      arrayed << yield(self[i])
+      i += 1
+
+    end
+
+    arrayed
+  end
+
+  def my_inject(val = nil, val2 = nil)
+    result = 0
+
+    if (val.is_a? Integer) && (val2.is_a? Symbol)
+      unshift(val)
+      loc = val2.to_s
+      my_inject { |summ, numberr| summ.method(loc).call(numberr) }
+    end
+
+    if val.is_a? Symbol
+      loc = val.to_s
+      my_inject { |summ, numberr| summ.method(loc).call(numberr) }
+
+    end
+
+    if (val.is_a? Integer) && block_given?
+
+      unshift(val) if val.is_a? Integer
+
+    end
+
+    i = 0
+
+    while i < size - 1
+      self[i + 1] = yield(self[i], self[i + 1]) if block_given?
+      result = self[i + 1] if i == size - 2
+      i += 1
+    end
+
+    result
+  end
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
[1, 2, 3].my_inject { |memo, num| memo + num } #should return 6
[1, 2, 3].my_inject(1) { |memo, num| memo + num } #should return 7
```
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(:+)  #should return 6
[1, 2, 3].my_inject(1, :+)  #should return 7
```
- inject can perform on range too.
___
Well, you can say that you are having the exact result. But your approach is updating the existing array, it should not. Take a look at the following code. And try to execute it in your main.rb
```ruby
ARRAY_SIZE = 10
LOWEST_VALUE = 1
HIGHEST_VALUE = 20
array = Array.new(ARRAY_SIZE) { rand(LOWEST_VALUE...HIGHEST_VALUE) }
operation = proc { |sum, n| sum + n }

p array.inject(&operation)
p array.my_inject(&operation)

print array
p array.my_inject(:+)
print array
p array.inject(:+)

p array.my_inject(3, :*) == array.inject(3, :*)
p array.inject(3, :*) == array.my_inject(3, :*)
```